File: bindings/php/src/ProvenSnmp.php
<?php
// SPDX-License-Identifier: PMPL-1.0-or-later
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// SNMP protocol types for proven-servers.

declare(strict_types=1);

namespace ProvenServers;

/** Version matching the Idris2 ABI tags. */
enum Version: int
{
    case V1 = 0;
    case V2c = 1;
    case V3 = 2;
}

/** PduType matching the Idris2 ABI tags. */
enum PduType: int
{
    case GetRequest = 0;
    case GetNextRequest = 1;
    case GetResponse = 2;
    case SetRequest = 3;
    case GetBulkRequest = 4;
    case InformRequest = 5;
    case SnmpV2Trap = 6;
}

/** ErrorStatus matching the Idris2 ABI tags. */
enum ErrorStatus: int
{
    case NoError = 0;
    case TooBig = 1;
    case NoSuchName = 2;
    case BadValue = 3;
    case ReadOnly = 4;
    case GenErr = 5;
    case NoAccess = 6;
    case WrongType = 7;
    case WrongLength = 8;
    case WrongValue = 9;
    case NoCreation = 10;
    case InconsistentValue = 11;
    case ResourceUnavailable = 12;
    case CommitFailed = 13;
    case UndoFailed = 14;
    case AuthorizationError = 15;
}

File: bindings/csharp/src/ProvenSnmp.cs
// SPDX-License-Identifier: PMPL-1.0-or-later
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// SNMP protocol types for proven-servers.

namespace Proven;

/// <summary>Version matching the Idris2 ABI tags (0-2).</summary>
public enum Version : byte
{
    V1 = 0,
    V2c = 1,
    V3 = 2
}

/// <summary>PduType matching the Idris2 ABI tags (0-6).</summary>
public enum PduType : byte
{
    GetRequest = 0,
    GetNextRequest = 1,
    GetResponse = 2,
    SetRequest = 3,
    GetBulkRequest = 4,
    InformRequest = 5,
    SnmpV2Trap = 6
}

/// <summary>ErrorStatus matching the Idris2 ABI tags (0-15).</summary>
public enum ErrorStatus : byte
{
    NoError = 0,
    TooBig = 1,
    NoSuchName = 2,
    BadValue = 3,
    ReadOnly = 4,
    GenErr = 5,
    NoAccess = 6,
    WrongType = 7,
    WrongLength = 8,
    WrongValue = 9,
    NoCreation = 10,
    InconsistentValue = 11,
    ResourceUnavailable = 12,
    CommitFailed = 13,
    UndoFailed = 14,
    AuthorizationError = 15
}
